lib/ts/Layout.cc

```diff
@@ -41,56 +47,61 @@ Layout::get()
 }
 
 void
-Layout::create(const char *prefix)
+Layout::create(ts::string_view prefix)
 {
   if (layout == nullptr) {
     layout = new Layout(prefix);
   }
 }
 
-static char *
-layout_relative(const char *root, const char *file)
+static bool
+_relative(char *path, ts::string_view root, ts::string_view file)
 {
-  char path[PATH_NAME_MAX];
-
-  if (ink_filepath_merge(path, PATH_NAME_MAX, root, file, INK_FILEPATH_TRUENAME)) {
+  if (ink_filepath_merge(path, PATH_NAME_MAX, root.data(), file.data(), INK_FILEPATH_TRUENAME)) {
     int err = errno;
     // Log error
     if (err == EACCES) {
-      ink_error("Cannot merge path '%s' above the root '%s'\n", file, root);
+      ink_error("Cannot merge path '%s' above the root '%s'\n", file.data(), root.data());
     } else if (err == E2BIG) {
       ink_error("Exceeding file name length limit of %d characters\n", PATH_NAME_MAX);
     } else {
       // TODO: Make some pretty errors.
-      ink_error("Cannot merge '%s' with '%s' error=%d\n", file, root, err);
+      ink_error("Cannot merge '%s' with '%s' error=%d\n", file.data(), root.data(), err);
     }
-    return nullptr;
+    return false;
+  }
+  return true;
+}
+
+static string
+layout_relative(ts::string_view root, ts::string_view file)
+{
+  char path[PATH_NAME_MAX];
+  string ret;
+  if (_relative(path, root, file)) {
+    ret = path;
   }
-  return ats_strdup(path);
+  return ret;
 }
 
 char *
-Layout::relative(const char *file)
+Layout::relative(ts::string_view file)
```

Comment:
This is because removing this goes down the rabbit hole of existing code that has to be fixed. I suggested we make it so we can make a different commit to remove the functions returning char* later as this would really make a huge diff of code to do it all at once